enhancements/microshift/low-latency-workloads-on-microshift.md

```diff
@@ -0,0 +1,612 @@
+  - "@pacevedom, MicroShift lead"
+approvers:
+  - "@jerpeter1"
+api-approvers:
+  - TBD
+creation-date: 2024-06-12
+last-updated: 2024-06-12
+tracking-link:
+  - USHIFT-2981
+---
+
+# Low Latency workloads on MicroShift
+
+## Summary
+
+This enhancement describes how low latency workloads will be supported on MicroShift hosts.
+
+
+## Motivation
+
+Some customers want to run latency sensitive workload like software defined PLCs.
+
+Currently it's possible, but requires substantial amount of knowledge to correctly configure
+all involved components whereas in OpenShift everything is abstracted using Node Tuning Operator's
+PerformanceProfile CR.
+Therefore, this enhancement focuses on making this configuration easy for customers by shipping
+ready to use packages to kickstart customers' usage of low latency workloads.
+
+
+### User Stories
+
+* As a MicroShift administrator, I want to configure MicroShift host,
+  so that I can run low latency workloads.
+
+
+### Goals
+
+Provide relatively easy way to configure system for low latency workload running on MicroShift:
+- Prepare low latency TuneD profile for MicroShift
+- Prepare necessary CRI-O configurations
+- Allow configuration of Kubelet via MicroShift config
+- Add small systemd daemon to enable TuneD profile and (optionally) reboot the host if the kernel
+  arguments change to make them effective
+  
+
+### Non-Goals
+
+- Workload partitioning (i.e. pinning MicroShift control plane components)
+- Duplicate all capabilities of Node Tuning Operator
+
+
+## Proposal
+
+To ease configuration of the system for running low latency workloads on MicroShift following
+parts need to be put in place:
+- TuneD profile
+- CRI-O configuration + Kubernetes' RuntimeClass
+- Kubelet configuration (CPU, Memory, and Topology Managers)
+- Small systemd daemon to activate TuneD profile on boot and reboot the host if the kernel args
+  are changed.
+
+New RPM will be created that will contain tuned profile, CRI-O configs, and mentioned systemd daemon.
+We'll leverage existing know how of Performance and Scalability team expertise and look at
+Node Tuning Operator capabilities.
+
+To allow customization of supplied TuneD profile for specific system, this new profile will
+include instruction to include file with variables, which can be overridden by the user.
+
+All of this will be accompanied by step by step documentation on how to use this feature,
+tweak values for specific system, and what are the possibilities and limitations.
+
+Optionally, a new subcommand `microshift doctor low-latency` might be added to main
+MicroShift binary to provide some verification checks if system configuration is matching
+expectations according to our knowledge. It shall not configure system - only report potential problems.
+
+
+### Workflow Description
+
+Workflow consists of two parts:
+1. system and MicroShift configuration
+1. preparing Kubernetes manifests for low latency
+
+#### System and MicroShift configuration
```

Comment:
Should we also add a bootc branch here? Or, at least a note on what it would look like.